api/v1alpha3/machinedeployment_webhook.go

```diff
@@ -78,3 +86,54 @@ func (m *MachineDeployment) validate() error {
 
 	return apierrors.NewInvalid(GroupVersion.WithKind("MachineDeployment").GroupKind(), m.Name, allErrs)
 }
+
+// PopulateDefaultsMachineDeployment fills in default field values
+// This is also called during MachineDeployment sync.
+func PopulateDefaultsMachineDeployment(d *MachineDeployment) {
+	if d.Spec.Replicas == nil {
+		d.Spec.Replicas = new(int32)
+		*d.Spec.Replicas = 1
```

Comment:
Can use `Int32Ptr` from `k8s.io/utils` for this instead: https://github.com/kubernetes/utils/blob/master/pointer/pointer.go#L50

Similar for defaulting the other pointer values as well.